apps/web/src/components/templates/BlueprintModal.tsx

```diff
@@ -0,0 +1,89 @@
+import { Modal, useMantineTheme } from '@mantine/core';
+import { colors, shadows, Title, Text, Button } from '../../design-system';
+import { Center, Loader } from '@mantine/core';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+import { createTemplateFromBluePrintId, getBlueprintTemplateById } from '../../api/templates';
+import { When } from '../utils/When';
+import { ActivePageEnum } from '../../pages/templates/editor/TemplateEditorPage';
+import { errorMessage } from '../../utils/notifications';
+
+export function BluePrintModal({ blueprintId }: { blueprintId?: string }) {
+  const theme = useMantineTheme();
+  const navigate = useNavigate();
+  const onClose = () => {
+    localStorage.removeItem('blueprintId');
+    navigate('/templates');
+  };
+
+  const { data: blueprint, isInitialLoading: isBluePrintLoading } = useQuery(
+    ['blueprint', blueprintId],
+    () => getBlueprintTemplateById(blueprintId as string),
+    {
+      enabled: !!blueprintId,
+      onError: onClose,
+    }
+  );
+
+  const { mutate, isLoading: isCreating } = useMutation(createTemplateFromBluePrintId, {
+    onSuccess: (template) => {
+      localStorage.removeItem('blueprintId');
+      if (template) {
+        navigate(`/templates/edit/${template?._id}?page=${ActivePageEnum.WORKFLOW}`);
+      }
+    },
+    onError: (err: any) => {
+      if (err?.message) {
+        errorMessage(err?.message);
+      }
+      onClose();
+    },
+  });
+
+  return (
+    <>
+      <Modal
+        opened={blueprintId !== undefined}
+        overlayColor={theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight}
+        overlayOpacity={0.7}
+        styles={{
+          modal: {
+            backgroundColor: theme.colorScheme === 'dark' ? colors.B15 : colors.white,
+          },
+          body: {
+            paddingTop: '5px',
+          },
+          inner: {
+            paddingTop: '180px',
+          },
+        }}
+        withCloseButton={!isBluePrintLoading && !isCreating}
+        closeOnClickOutside={!isBluePrintLoading && !isCreating}
```

Comment:
Not sure if this is not right but for both properties conditions are the same. If it is meant to be like that maybe we should keep that condition in a function or variable and reuse it in both.